### PR TITLE
add a directory for domain-cc and a couple requirements

### PR DESCRIPTION
**Pull request by lukey-luke**
## What was the problem?
Contention classification team needs a place to put source code.

Associated tickets or Slack threads:
- [VA.gov Claim Classification set up directory task](https://github.com/orgs/department-of-veterans-affairs/projects/873#)

## How does this fix it?[^1]
Developer can clone and run sample endpoint.

## How to test this PR
- Just getting started on a prototype, this probably doesn't merit testing
- Follow instructions in the [CC readme](https://github.com/department-of-veterans-affairs/abd-vro/blob/feature-add-domain-cc/domain-cc/README.md) if you want to run fastAPI for fun

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Comment by lukey-luke** [comment]
Yoom provided some great feedback w/ an example domain. I think we should keep this in mind.
See [@Tag change](https://github.com/department-of-veterans-affairs/abd-vro/pull/1213/files#diff-e24c056b11916bcacf0934a4e5ec8bb6e38ae4106c7324093181935145cf9b20R57) required to link docs to parent doc.

I did include a readme w/ instructions on setup + running and I used a separate pre-commit-config so we can have a little bit stricter formatting w/ [black](https://black.readthedocs.io/en/stable/).

I hope this is enough to get my team unblocked and we can start contributing.

File: domain-cc/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Run pre-commit against all files by executing: pre-commit run --all-files
```
Comment (yoomlam) [inline domain-cc/.pre-commit-config.yaml:1]: Why is this file here?

File: domain-cc/.pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    hooks: # https://github.com/pre-commit/pre-commit-hooks#hooks-available
+      - id: check-merge-conflict
+      - id: check-executables-have-shebangs
+      - id: check-shebang-scripts-are-executable
+        exclude: ^scripts/.*\.src$
+      - id: check-yaml
+        exclude: ^.*helm-.*$
+      - id: destroyed-symlinks
+      - id: detect-private-key
+        exclude: ^svc-lighthouse-api/.*/test/resources/application-test.properties$
```
Comment (yoomlam) [inline domain-cc/.pre-commit-config.yaml:30]: This line is irrelevant to you
```suggestion
```

File: domain-cc/.pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Run pre-commit against all files by executing: pre-commit run --all-files
+# Auto-update the config to the latest repos' versions by executing: pre-commit autoupdate
+# Rather than disabling particular hooks (https://pre-commit.com/#temporarily-disabling-hooks), add an `exclude`
+
+# https://pre-commit.com/#regular-expressions
+exclude: >
+  (?x)^(
+      app/src/docker/docker-compose.yml
+  )
```
Comment (yoomlam) [inline domain-cc/.pre-commit-config.yaml:9]: Irrelevant. Leaving as a comment for reference:
```suggestion
# exclude: >
#  (?x)^(
#      app/src/docker/docker-compose.yml
#  )
```
or you can delete it.

**Review by yoomlam — APPROVED** [review]

File: domain-cc/.pre-commit-config.yaml (hunk 3 above)
Comment (lukey-luke) [reply domain-cc/.pre-commit-config.yaml]: deleted

File: domain-cc/.pre-commit-config.yaml (hunk 2 above)
Comment (lukey-luke) [reply domain-cc/.pre-commit-config.yaml]: deleted

**Review by carly-jones — APPROVED** [review]
